helix-term/src/commands/typed.rs

```diff
@@ -1426,19 +1426,28 @@ fn lsp_workspace_command(
         };
         cx.jobs.callback(callback);
     } else {
-        let command = args.join(" ");
+        let command = args.first().unwrap().to_string();
+
         let matches: Vec<_> = ls_id_commands
             .filter(|(_ls_id, c)| *c == &command)
             .collect();
 
         match matches.as_slice() {
             [(ls_id, _command)] => {
+                let lsp_command_args: Vec<Value> = args[1..]
+                    .iter()
+                    .map(|s| Value::String(s.to_string()))
+                    .collect();
```

Comment:
If we're parsing a list of JSON values then we should use the streaming deserializer from serde_json like https://github.com/helix-editor/helix/pull/12527 does for `:toggle-option`. The streaming deserializer will take a slightly different syntax for strings - `:lsp-workspace-command lsp.Command "foo" "bar"` - which would be a breaking change to the syntax added if we merge this as-is.

I'll include this change on that PR since it's not possible to parse the args accurately with the way we parse the command line on master